components/script/dom/html/htmlhyperlinkelementutils.rs

```diff
@@ -75,24 +75,22 @@ impl<T: HyperlinkElement + DerivedFrom<Element> + Castable + NodeTraits> Hyperli
         self.reinitialize_url();
 
         // Step 2. Let url be this's url.
-        let url = match self.get_url().borrow_mut().as_mut() {
-            // Step 3. If url is null, then return.
-            None => return,
-            // Step 4. If the given value is the empty string, set url's fragment to null.
-            // Note this step is taken care of by UrlHelper::SetHash when the value is Some
-            // Steps 5. Otherwise:
-            Some(url) => {
-                // Step 5.1. Let input be the given value with a single leading "#" removed, if any.
-                // Step 5.2. Set url's fragment to the empty string.
-                // Note these steps are taken care of by UrlHelper::SetHash
-                UrlHelper::SetHash(url, value);
-
-                // Step 5.4.  Basic URL parse input, with url as url and fragment state as state
-                // override.
-                DOMString::from(url.as_str())
-            },
+        // Step 3. If url is null, then return.
+        let mut url = self.get_url().borrow_mut();
+        let Some(url) = url.as_mut() else {
+            return;
         };
 
+        // Step 4. If the given value is the empty string, set url's fragment to null.
+        // Note this step is taken care of by UrlHelper::SetHash when the value is Some
+        // Steps 5. Otherwise:
+        // Step 5.1. Let input be the given value with a single leading "#" removed, if any.
+        // Step 5.2. Set url's fragment to the empty string.
+        // Note these steps are taken care of by UrlHelper::SetHash
+        UrlHelper::SetHash(url, value);
+
+        // Step 5.4.  Basic URL parse input, with url as url and fragment state as state
+        // override.
```

Comment:
Nit: The rough structure of all these algorithms is the same, and the spec comment that updates the relevant segment of the URL (in this case 5.4) is usually above the call to `UrlHelper::SetXYZ`. In this function its above the call to `update_href` instead.

Reply:
Thanks. I've fixed this.